ext/spl/spl_iterators.c

```diff
@@ -560,10 +558,18 @@ static void spl_recursive_it_it_construct(INTERNAL_FUNCTION_PARAMETERS, zend_cla
 				Z_ADDREF_P(iterator);
 			}
 
-			ZVAL_LONG(&caching_it_flags, user_caching_it_flags);
-			spl_instantiate_arg_ex2(spl_ce_RecursiveCachingIterator, &caching_it, iterator, &caching_it_flags);
-			zval_ptr_dtor(&caching_it_flags);
+			/* Instantiate object and call constructor */
+			zval params[2];
+			ZVAL_COPY(&params[0], iterator); // TODO Prevent copy?
```

Comment:
ZVAL_COPY_VALUE and remove the line below?
But the refcounting here for `iterator` looks a bit iffy.